From f30ad91b99f1f99d5397a9547969f8a144aeb035 Mon Sep 17 00:00:00 2001
From: Haoran Wang <elven.wang@nxp.com>
Date: Wed, 16 Mar 2022 03:02:31 +0800
Subject: [PATCH 1/2] Add Network Commissioning cluster setup for lighting app
 Linux

For issue #15261 the Linux lighting app failed to be commission by
chip-tool pairing ble-wifi command.
Based on the message in #14751, the Linux platform intented to support
WiFi network by default instead Thread.

This patch add Network Commissioning cluster setup for WiFi device for
lighting-app.

Signed-off-by: Haoran Wang <elven.wang@nxp.com>
---
 examples/lighting-app/linux/main.cpp | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/examples/lighting-app/linux/main.cpp b/examples/lighting-app/linux/main.cpp
index d3a07fb97..74816f1c7 100644
--- a/examples/lighting-app/linux/main.cpp
+++ b/examples/lighting-app/linux/main.cpp
@@ -22,15 +22,25 @@
 #include <app-common/zap-generated/ids/Attributes.h>
 #include <app-common/zap-generated/ids/Clusters.h>
 #include <app/ConcreteAttributePath.h>
+#include <app/clusters/network-commissioning/network-commissioning.h>
 #include <lib/support/logging/CHIPLogging.h>
+#include <platform/Linux/NetworkCommissioningDriver.h>
 
 #if defined(PW_RPC_ENABLED)
 #include "Rpc.h"
 #endif // PW_RPC_ENABLED
 
 using namespace chip;
+using namespace chip::app;
 using namespace chip::app::Clusters;
 
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+namespace {
+DeviceLayer::NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(0, &sLinuxWiFiDriver);
+} // namespace
+#endif
+
 void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath & attributePath, uint8_t mask, uint8_t type,
                                        uint16_t size, uint8_t * value)
 {
@@ -60,7 +70,12 @@ void emberAfOnOffClusterInitCallback(EndpointId endpoint)
     // TODO: implement any additional Cluster Server init actions
 }
 
-void ApplicationInit() {}
+void ApplicationInit()
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+    sWiFiNetworkCommissioningInstance.Init();
+#endif
+}
 
 int main(int argc, char * argv[])
 {
-- 
2.25.1

